kmk/kmk_keyboard.py

```diff
@@ -67,32 +67,17 @@ class KMKKeyboard:
     # real known fix yet other than turning off debug, but M4s have always been
     # tight on RAM so....
     def __repr__(self):
-        return (
-            'KMKKeyboard('
-            'debug_enabled={} '
-            'diode_orientation={} '
-            'matrix={} '
-            'unicode_mode={} '
-            '_hid_helper={} '
-            'keys_pressed={} '
-            'coordkeys_pressed={} '
-            'hid_pending={} '
-            'active_layers={} '
-            'timeouts={} '
-            ')'
-        ).format(
-            self.debug_enabled,
-            self.diode_orientation,
-            self.matrix,
-            self.unicode_mode,
-            self._hid_helper,
-            # internal state
-            self.keys_pressed,
-            self._coordkeys_pressed,
-            self.hid_pending,
-            self.active_layers,
-            self._timeouts,
+        txt = 'KMKKeyboard(\n'
+        txt += '  debug_enabled={}, diode_orientation={}, matrix={}\n'.format(
+            self.debug_enabled, self.diode_orientation, self.matrix
         )
+        txt += f'  unicode_mode={self.unicode_mode}, _hid_helper={self._hid_helper},\n'
+        txt += f'  keys_pressed={self.keys_pressed}\n'
+        txt += f'  _coordkeys_pressed={self._coordkeys_pressed}\n'
+        txt += '  hid_pending={}, active_layers={}, _timeouts={}\n)'.format(
```

Comment:
Why is this not an f-string but the others are?

Reply:
those are not fstring as that would make the line too long. The goal of fstring is to give an option that in most case is  a more readable string. Here the more readable, given the lenght, is the format. as it is  each argument is below it's corresponding {}.
It was also an attempt to have both not too long line, and "the output will look exactly like the string is formated in the code" you suggested.

I can split it into more fstring line if the consistency of string formatting is wanted. (I can even push a commit to remove all `.format()` given the low number of them if you want).

Reply:
I'll leave the final call to @xs5871 but I believe that consistency makes more sense personally, but I suppose that's a valid point.

Reply:
+1 to consistency.
When I suggested "make it an f-string" I had a single multi-line f-string in mind. The `x += '*\n'` feels quite unpythonic.

I have to say, this discussion has been going on for almost to long, considering this concerns such a minor change.
It's probably because we started with a complex, possibly over-engineered (no offense, we all do over-engineer sometimes; also it's fun), solution and step for step reverted it back to a pragmatic one, instead of starting from scratch.
At this point, we could just as well keep the original code, just with a couple of newlines thrown into it.